src/tactic/interactive.lean

```diff
@@ -635,5 +652,90 @@ do let ns := name_set.of_list xs,
      when (¬ ns.contains h.local_pp_name) $
        try $ tactic.clear h) ∘ list.reverse
 
+meta def format_names (ns : list name) : format :=
+format.join $ list.intersperse " " (ns.map to_fmt)
+
+private meta def format_binders : list name × binder_info × expr → tactic format
+| (ns, binder_info.default, t) := pformat!"({format_names ns} : {t})"
+| (ns, binder_info.implicit, t) := pformat!"{{{format_names ns} : {t}}"
+| (ns, binder_info.strict_implicit, t) := pformat!"⦃{format_names ns} : {t}⦄"
+| (ns, binder_info.inst_implicit, t) := pformat!"[{format_names ns} : {t}]"
+| (ns, binder_info.aux_decl, t) := pformat!"({format_names ns} : {t})"
+
+meta def mk_paragraph_aux (right_margin : ℕ) : format → format → ℕ → list format → format
+| par ln len [] := par ++ format.line ++ ln
+| par ln len (x :: xs) :=
+  let len' := x.to_string.length in
+  if len + len' ≤ right_margin then
+    mk_paragraph_aux par (ln ++ x ++ " ") (len + len' + 1) xs
+  else
+    mk_paragraph_aux (par ++ format.line ++ ln) ("  " ++ x ++ " ") len' xs
+
+meta def mk_paragraph (right_margin : ℕ) : list format → format :=
```

Comment:
Doc string? This seems usable elsewhere but it isn't clear what exactly it does.